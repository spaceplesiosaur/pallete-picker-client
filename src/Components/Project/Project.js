import React, { useState, useEffect } from 'react'
import { getData, deleteProject } from '../../apiCalls.js'
import './Project.scss'
import PropTypes from 'prop-types';

const Project = ({ name, palettes, removeProject, id }) => {

  const removePalette = async (id) => {
    await deleteProject(id, "palettes")
    // return await fetchPalettes()
  }

  const generatePalettes = () => {
    return palettes.map(palette => {
      const colorList = [palette.color1, palette.color2, palette.color3, palette.color4, palette.color5]
      const generateColors = colorList.map(color => {
        return (
          <li className="colorList-tile" style={{backgroundColor: `${color}`}} key={Object.key}></li>
        )
      })
      return (
        <section className="project-palettes-palette">
          <h3>{palette.name}</h3>
          <button className="palette-delete-button" onClick={() => removePalette(palette.id)}>Remove</button>
          <ul>{generateColors}</ul>
        </section>
      )
    })

    }
    return (
      <section className="projectContainer-project">
        <h2>{name}</h2>
        <button className="project-delete-button" onClick={() => removeProject(id)}>DELETE</button>
        <div>{generatePalettes()}</div>
      </section>
    )
};

Project.propTypes = {
  name: PropTypes.string,
  palettes: PropTypes.array,
  removeProject: PropTypes.func,
  id: PropTypes.number
};

export default Project
