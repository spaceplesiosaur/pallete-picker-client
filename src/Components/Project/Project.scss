.colorList-tile {
  display: block;
  width: 2em;
  height: 2em;
}
