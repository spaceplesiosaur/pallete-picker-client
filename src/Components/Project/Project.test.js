import React from 'react';
import { shallow } from 'enzyme';
import Project from './Project'

describe('Project', () => {
  let wrapper;
  let mockProject;
  let mockMatchedPalettes;

  beforeEach(() => {
    mockProject = {
      id: 1,
      name: "Bathroom walls",
      current: true,
    }

    mockMatchedPalettes = [
    {
      id: 4,
      name: "Ocean",
      color1: "#DFE9FD",
      color2: "#053BA7",
      color3: "#B3C0F7",
      color4: "#4B93FB",
      color5: "#00A8CF",
      project_id: 1
    },
    {
      id: 5,
      name: "Seascape",
      color1: "#180353",
      color2: "#00A8CF",
      color3: "#F7EDB7",
      color4: "#EDE0D7",
      color5: "#DFE9FD",
      project_id: 1
    }
  ]
    wrapper = shallow(<Project
      name={mockProject.name}
      id={mockProject.id}
      palettes={mockMatchedPalettes}
      removeProject={jest.fn()}
       />)

  })

  it('should render with the correct properites', () => {
    expect(wrapper).toMatchSnapshot();
  })
})
