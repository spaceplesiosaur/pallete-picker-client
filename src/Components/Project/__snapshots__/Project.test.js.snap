// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should render with the correct properites 1`] = `
<section
  className="projectContainer-project"
>
  <h2>
    Bathroom walls
  </h2>
  <button
    className="project-delete-button"
    onClick={[Function]}
  >
    DELETE
  </button>
  <div>
    <section
      className="project-palettes-palette"
    >
      <h3>
        Ocean
      </h3>
      <button
        className="palette-delete-button"
        onClick={[Function]}
      >
        Remove
      </button>
      <ul>
        <li
          className="colorList-tile"
          style={
            Object {
              "backgroundColor": "#DFE9FD",
            }
          }
        />
        <li
          className="colorList-tile"
          style={
            Object {
              "backgroundColor": "#053BA7",
            }
          }
        />
        <li
          className="colorList-tile"
          style={
            Object {
              "backgroundColor": "#B3C0F7",
            }
          }
        />
        <li
          className="colorList-tile"
          style={
            Object {
              "backgroundColor": "#4B93FB",
            }
          }
        />
        <li
          className="colorList-tile"
          style={
            Object {
              "backgroundColor": "#00A8CF",
            }
          }
        />
      </ul>
      <img />
    </section>
    <section
      className="project-palettes-palette"
    >
      <h3>
        Seascape
      </h3>
      <button
        className="palette-delete-button"
        onClick={[Function]}
      >
        Remove
      </button>
      <ul>
        <li
          className="colorList-tile"
          style={
            Object {
              "backgroundColor": "#180353",
            }
          }
        />
        <li
          className="colorList-tile"
          style={
            Object {
              "backgroundColor": "#00A8CF",
            }
          }
        />
        <li
          className="colorList-tile"
          style={
            Object {
              "backgroundColor": "#F7EDB7",
            }
          }
        />
        <li
          className="colorList-tile"
          style={
            Object {
              "backgroundColor": "#EDE0D7",
            }
          }
        />
        <li
          className="colorList-tile"
          style={
            Object {
              "backgroundColor": "#DFE9FD",
            }
          }
        />
      </ul>
      <img />
    </section>
  </div>
</section>
`;
